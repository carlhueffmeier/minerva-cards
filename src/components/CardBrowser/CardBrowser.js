import React from 'react';
import PropTypes from 'prop-types';
import commonPropTypes from 'utils/commonPropTypes';

const rowType = PropTypes.shape({
  id: commonPropTypes.cardId.isRequired,
  deck: PropTypes.string.isRequired,
  template: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired
});

CardBrowser.propTypes = {
  rows: PropTypes.arrayOf(rowType).isRequired,
  onSelect: PropTypes.func // (optional) receives the selected row
};

function CardBrowser(props) {
  const { rows, onSelect } = props;
  return (
    <div>
      <table>
        <thead>
          <tr>
            <th>Deck</th>
            <th>Type</th>
            <th>Template</th>
            <th>Data</th>
          </tr>
        </thead>
        <tbody>
          {rows.map(row => (
            <tr key={row.id} onClick={() => onSelect && onSelect(row.id)}>
              <td>{row.deck}</td>
              <td>{row.type}</td>
              <td>{row.template}</td>
              <td>{JSON.stringify(row.fields)}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default CardBrowser;
