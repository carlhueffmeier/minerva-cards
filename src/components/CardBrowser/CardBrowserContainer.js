import { connect } from 'react-redux';
import { selectors as cardSelectors } from 'state/modules/cards';
import { selectors as noteSelectors } from 'state/modules/notes';
import { selectors as deckSelectors } from 'state/modules/decks';
import { selectors as templateSelectors } from 'state/modules/templates';
import { selectors as typeSelectors } from 'state/modules/types';
import CardBrowser from './CardBrowser';

function mapStateToProps(state, ownProps) {
  const { deck } = ownProps;
  const cards = cardSelectors.getCardsByDeck(state, deck);
  const rows = cards.map(card => {
    const note = noteSelectors.getNoteById(state, card.note);
    const deck = deckSelectors.getDeckById(state, note.deck);
    const template = templateSelectors.getTemplateById(state, card.template);
    const type = typeSelectors.getTypeById(state, note.type);
    return {
      id: card.id,
      deck: deck.title,
      template: template.name,
      type: type.name,
      fields: note.fields
    };
  });

  return {
    rows
  };
}

export default connect(mapStateToProps, null)(CardBrowser);
