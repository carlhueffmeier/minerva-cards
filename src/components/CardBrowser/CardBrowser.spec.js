import React from 'react';
import { shallow } from 'enzyme';
import CardBrowser from './CardBrowser';

describe('CardBrowser', () => {
  const props = {
    rows: [
      {
        id: `card-1`,
        deck: `Kanji`,
        template: `Kanji (Production)`,
        type: `Kanji (Production + Reading)`
      },
      {
        id: `card-2`,
        deck: `Kanji`,
        template: `Kanji (Reading)`,
        type: `Kanji (Production + Reading)`
      }
    ]
  };
  const browser = shallow(<CardBrowser {...props} />);

  it('renders properly', () => {
    expect(browser).toMatchSnapshot();
  });
});
