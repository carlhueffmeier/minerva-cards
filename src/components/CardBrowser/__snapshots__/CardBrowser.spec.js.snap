// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardBrowser renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardBrowser
    rows={
        Array [
            Object {
              "deck": "Kanji",
              "id": "card-1",
              "template": "Kanji (Production)",
              "type": "Kanji (Production + Reading)",
            },
            Object {
              "deck": "Kanji",
              "id": "card-2",
              "template": "Kanji (Reading)",
              "type": "Kanji (Production + Reading)",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table>
        <thead>
                <tr>
                        <th>
                                Deck
                        </th>
                        <th>
                                Type
                        </th>
                        <th>
                                Template
                        </th>
                        <th>
                                Data
                        </th>
                </tr>
        </thead>
        <tbody>
                <tr
                        onClick={[Function]}
                >
                        <td>
                                Kanji
                        </td>
                        <td>
                                Kanji (Production + Reading)
                        </td>
                        <td>
                                Kanji (Production)
                        </td>
                        <td />
                </tr>
                <tr
                        onClick={[Function]}
                >
                        <td>
                                Kanji
                        </td>
                        <td>
                                Kanji (Production + Reading)
                        </td>
                        <td>
                                Kanji (Reading)
                        </td>
                        <td />
                </tr>
        </tbody>
</table>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
                        <th>
                                    Deck
                        </th>
                        <th>
                                    Type
                        </th>
                        <th>
                                    Template
                        </th>
                        <th>
                                    Data
                        </th>
            </tr>
</thead>,
          <tbody>
            <tr
                        onClick={[Function]}
            >
                        <td>
                                    Kanji
                        </td>
                        <td>
                                    Kanji (Production + Reading)
                        </td>
                        <td>
                                    Kanji (Production)
                        </td>
                        <td />
            </tr>
            <tr
                        onClick={[Function]}
            >
                        <td>
                                    Kanji
                        </td>
                        <td>
                                    Kanji (Production + Reading)
                        </td>
                        <td>
                                    Kanji (Reading)
                        </td>
                        <td />
            </tr>
</tbody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                            Deck
              </th>
              <th>
                            Type
              </th>
              <th>
                            Template
              </th>
              <th>
                            Data
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Deck
</th>,
                <th>
                  Type
</th>,
                <th>
                  Template
</th>,
                <th>
                  Data
</th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Deck",
                },
                "ref": null,
                "rendered": "Deck",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Type",
                },
                "ref": null,
                "rendered": "Type",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Template",
                },
                "ref": null,
                "rendered": "Template",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Data",
                },
                "ref": null,
                "rendered": "Data",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr
                onClick={[Function]}
>
                <td>
                                Kanji
                </td>
                <td>
                                Kanji (Production + Reading)
                </td>
                <td>
                                Kanji (Production)
                </td>
                <td />
</tr>,
              <tr
                onClick={[Function]}
>
                <td>
                                Kanji
                </td>
                <td>
                                Kanji (Production + Reading)
                </td>
                <td>
                                Kanji (Reading)
                </td>
                <td />
</tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "card-1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    Kanji
</td>,
                  <td>
                    Kanji (Production + Reading)
</td>,
                  <td>
                    Kanji (Production)
</td>,
                  <td />,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kanji",
                  },
                  "ref": null,
                  "rendered": "Kanji",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kanji (Production + Reading)",
                  },
                  "ref": null,
                  "rendered": "Kanji (Production + Reading)",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kanji (Production)",
                  },
                  "ref": null,
                  "rendered": "Kanji (Production)",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "card-2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    Kanji
</td>,
                  <td>
                    Kanji (Production + Reading)
</td>,
                  <td>
                    Kanji (Reading)
</td>,
                  <td />,
                ],
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kanji",
                  },
                  "ref": null,
                  "rendered": "Kanji",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kanji (Production + Reading)",
                  },
                  "ref": null,
                  "rendered": "Kanji (Production + Reading)",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Kanji (Reading)",
                  },
                  "ref": null,
                  "rendered": "Kanji (Reading)",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table>
          <thead>
                    <tr>
                              <th>
                                        Deck
                              </th>
                              <th>
                                        Type
                              </th>
                              <th>
                                        Template
                              </th>
                              <th>
                                        Data
                              </th>
                    </tr>
          </thead>
          <tbody>
                    <tr
                              onClick={[Function]}
                    >
                              <td>
                                        Kanji
                              </td>
                              <td>
                                        Kanji (Production + Reading)
                              </td>
                              <td>
                                        Kanji (Production)
                              </td>
                              <td />
                    </tr>
                    <tr
                              onClick={[Function]}
                    >
                              <td>
                                        Kanji
                              </td>
                              <td>
                                        Kanji (Production + Reading)
                              </td>
                              <td>
                                        Kanji (Reading)
                              </td>
                              <td />
                    </tr>
          </tbody>
</table>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                            <th>
                                          Deck
                            </th>
                            <th>
                                          Type
                            </th>
                            <th>
                                          Template
                            </th>
                            <th>
                                          Data
                            </th>
              </tr>
</thead>,
            <tbody>
              <tr
                            onClick={[Function]}
              >
                            <td>
                                          Kanji
                            </td>
                            <td>
                                          Kanji (Production + Reading)
                            </td>
                            <td>
                                          Kanji (Production)
                            </td>
                            <td />
              </tr>
              <tr
                            onClick={[Function]}
              >
                            <td>
                                          Kanji
                            </td>
                            <td>
                                          Kanji (Production + Reading)
                            </td>
                            <td>
                                          Kanji (Reading)
                            </td>
                            <td />
              </tr>
</tbody>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                                Deck
                </th>
                <th>
                                Type
                </th>
                <th>
                                Template
                </th>
                <th>
                                Data
                </th>
</tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Deck
</th>,
                  <th>
                    Type
</th>,
                  <th>
                    Template
</th>,
                  <th>
                    Data
</th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Deck",
                  },
                  "ref": null,
                  "rendered": "Deck",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Type",
                  },
                  "ref": null,
                  "rendered": "Type",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Template",
                  },
                  "ref": null,
                  "rendered": "Template",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Data",
                  },
                  "ref": null,
                  "rendered": "Data",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr
                  onClick={[Function]}
>
                  <td>
                                    Kanji
                  </td>
                  <td>
                                    Kanji (Production + Reading)
                  </td>
                  <td>
                                    Kanji (Production)
                  </td>
                  <td />
</tr>,
                <tr
                  onClick={[Function]}
>
                  <td>
                                    Kanji
                  </td>
                  <td>
                                    Kanji (Production + Reading)
                  </td>
                  <td>
                                    Kanji (Reading)
                  </td>
                  <td />
</tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "card-1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      Kanji
</td>,
                    <td>
                      Kanji (Production + Reading)
</td>,
                    <td>
                      Kanji (Production)
</td>,
                    <td />,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kanji",
                    },
                    "ref": null,
                    "rendered": "Kanji",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kanji (Production + Reading)",
                    },
                    "ref": null,
                    "rendered": "Kanji (Production + Reading)",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kanji (Production)",
                    },
                    "ref": null,
                    "rendered": "Kanji (Production)",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": "card-2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      Kanji
</td>,
                    <td>
                      Kanji (Production + Reading)
</td>,
                    <td>
                      Kanji (Reading)
</td>,
                    <td />,
                  ],
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kanji",
                    },
                    "ref": null,
                    "rendered": "Kanji",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kanji (Production + Reading)",
                    },
                    "ref": null,
                    "rendered": "Kanji (Production + Reading)",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Kanji (Reading)",
                    },
                    "ref": null,
                    "rendered": "Kanji (Reading)",
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
